### PR TITLE
My first game, man!

### PR DESCRIPTION
**Pull request by Jameshudson**

### CODE REVIEW

File: src/actors/ball.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import * as ex from 'excalibur';
+import {GameEvent} from "excalibur";
```
Comment (podpoint) [inline src/actors/ball.ts]: In terms of style generally have a space `{ GameEvent }`

Comment (mstuessy) [inline src/actors/ball.ts]: In terms of style generally encouraged to have space in single line curly braces `{}` like `{ GameEvent }`

Comment (mstuessy) [inline src/actors/ball.ts]: Used single quotes for string above, then double quotes here. Single quotes here I'd say

File: src/actors/ball.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+export class Ball extends ex.Actor {
+
+    constructor(public engine: ex.Engine) {
+        super({
+            pos: new ex.Vector(100, 300),
```
Comment (mstuessy) [inline src/actors/ball.ts]: feels like this could be a class or a type

File: src/scenes/level-one/level-one.ts [hunk 3]
```diff
@@ -1,7 +1,62 @@
 import * as ex from 'excalibur';
+import { Paddle } from '../../actors/player/paddle';
+import { Ball } from '../../actors/ball';
+import { Brick } from '../../actors/brick';
+import * as Events from "excalibur/dist/Events";
 
 export class LevelOne extends ex.Scene {
-  public onInitialize(engine: ex.Engine) {}
+
```
Comment (mstuessy) [inline src/scenes/level-one/level-one.ts]: 2 space tabs here, better to have 4 I think

File: src/actors/brick.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import * as ex from 'excalibur';
+import {GameEvent} from "excalibur";
+
+export class Brick extends ex.Actor {
+
+    constructor(pos: ex.Vector, width: number, height: number) {
+        super({
+            pos: pos,
```
Comment (mstuessy) [inline src/actors/brick.ts]: dry

File: src/actors/player/paddle.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import * as ex from 'excalibur';
+
+export class Paddle extends ex.Actor {
+    constructor(pos: ex.Vector) {
+        super({
+            pos: pos,
```
Comment (mstuessy) [inline src/actors/player/paddle.ts]: Same as above.

File: src/scenes/level-one/level-one.ts [hunk 6]
```diff
@@ -1,7 +1,62 @@
...
+        tempBrick.color = this._brickColours[j % this._brickColours.length];
+        this.bricks.push(tempBrick);
+        engine.add(tempBrick);
+      }
+    }
+
+    this.on('brickRemove', (env) => {
+      this.score += 10;
+    });
+  }
```
Comment (mstuessy) [inline src/scenes/level-one/level-one.ts]: a bit much going on in here I think, would help readability to break it up 👍

File: src/scenes/level-one/level-one.ts [hunk 7]
```diff
@@ -1,7 +1,62 @@
...
+    ex.Color.Violet,
+    ex.Color.Orange,
+    ex.Color. Yellow
+  ];
+
+  public onInitialize(engine: ex.Engine) {
+
+    const paddle = new Paddle(new ex.Vector(150,engine.drawHeight - 40));
+
+    engine.input.pointers.primary.on('move', function(evt) {
```
Comment (mstuessy) [inline src/scenes/level-one/level-one.ts]:
```
  ...
  ... on('move', evt => {
  ...
```

File: src/scenes/level-one/level-one.ts [hunk 8]
```diff
@@ -1,7 +1,62 @@
...
 export class LevelOne extends ex.Scene {
-  public onInitialize(engine: ex.Engine) {}
+
+  public bricks: Array<Brick> = [];
+
+  private  _padding: number = 20;
+  private _xoffset: number = 100;
+  private _yoffset: number = 20;
+  private _columns: number = 5;
+  private _rows: number = 3;
```
Comment (mstuessy) [inline src/scenes/level-one/level-one.ts]: I think linter will say that there should be spaces between class variables, not 100% sure though

File: src/scenes/level-one/level-one.ts [hunk 9]
```diff
@@ -1,7 +1,62 @@
...
+    this.add(paddle);
+    this.add(ball);
+
+    let brickWidth = engine.drawWidth / this._columns - this._padding - this._padding / this._columns;
+
+    for (let j = 0; j < this._rows; j++) {
+      for (let i = 0; i < this._columns; i++) {
+        let tempBrick = new Brick(new ex.Vector(
+            this._xoffset + i * (brickWidth + this._padding) + this._padding,
+            this._yoffset + j * (30 + this._padding) + this._padding
```
Comment (mstuessy) [inline src/scenes/level-one/level-one.ts]: Some of these numbers seem related, would help to make them class variables

File: src/scenes/level-one/level-one.ts [hunk 10]
```diff
@@ -1,7 +1,62 @@
...
+    });
+
+    let ball = new Ball(engine);
+    let scoreLabel = new ex.Label('Total score: ' + this.score, 50, 50);
+
+    this.add(scoreLabel);
+    this.add(paddle);
+    this.add(ball);
+
+    let brickWidth = engine.drawWidth / this._columns - this._padding - this._padding / this._columns;
```
Comment (mstuessy) [inline src/scenes/level-one/level-one.ts]: is brickWidth being reassigned?

File: src/actors/ball.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        ctx.fill()
+    }
+
+    private handleCollision(ev) {
+        if (this.engine.currentScene.bricks.indexOf(ev.other) > -1) {
+            ev.other.kill()
+            this.engine.currentScene.emit('brickRemove', ev.other);
+        }
+
+        let intersection = ev.intersection.normalize()
```
Comment (mstuessy) [inline src/actors/ball.ts]: intersection not being reassigned

File: src/actors/brick.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import * as ex from 'excalibur';
+import {GameEvent} from "excalibur";
```
Comment (mstuessy) [inline src/actors/brick.ts]: space

File: src/actors/brick.ts (hunk 4 above)
Comment (Jameshudson) [reply src/actors/brick.ts]: Doesn't make ts a bit useless?

**Pull request merged**